test/test_common/utility.h

```diff
@@ -570,6 +580,28 @@ MATCHER_P(ProtoEq, expected, "") {
   return equal;
 }
 
+MATCHER_P2(ProtoEqIgnoringField, expected, ignored_field, "") {
+  const bool equal = TestUtility::protoEqualIgnoringField(arg, expected, ignored_field);
+  if (!equal) {
+    int line_fill_len =
+        (68 - (std::string(ignored_field).length() + std::string("(but ignoring )").length()));
+    int first_half_len = line_fill_len / 2;
+    int second_half_len = line_fill_len - first_half_len;
+    std::string first_half(first_half_len, '=');
+    std::string second_half(second_half_len, '=');
```

Comment:
Do we need this ASCII art? It's OK if it helps make things much clearer, but the fewer lines dedicated to drawing ASCII art when trying to debug/understand code the better IMHO.

Reply:
It's
```
==========================Expected proto:===========================
===================(but ignoring response_nonce)====================
type_url: "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment"
response_nonce: "nonce"
------------------is not equal to actual proto:---------------------
type_url: "type.googleapis.com/envoy.api.v2.Listener"
response_nonce: "nonce"
====================================================================
```
vs
```
==========================Expected proto:===========================
=====(but ignoring response_nonce)=====
type_url: "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment"
response_nonce: "nonce"
------------------is not equal to actual proto:---------------------
type_url: "type.googleapis.com/envoy.api.v2.Listener"
response_nonce: "nonce"
====================================================================
```
I think it's reasonably readable but shabby looking the simpler way. (Not having any =s would in fact make it unreadable, I think). I split that code out into a helper; maybe it's cleaner this way? Actually, this new helper could be used in all of these matcher debug strings now, although I think it's clearer to stick with the hardcoded strings where possible.

Reply:
OK, if you want to make this a utility shared by all that's fine.

Reply:
Done.

Reply:
Unfortunately, with the helper used everywhere, clang format has decided to make it a little hideous.